File: src/main/java/scrumweb/dto/project/ProjectDto.java
package scrumweb.dto.project;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import scrumweb.dto.user.UserProfileDto;

import java.util.Set;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor
@Builder
@Getter @Setter
public class ProjectDto {
    private String name;
    private String projectKey;
    private String description;
    private UserProfileDto owner;
    private String icon;
    private Set<ProjectMemberDto> members;
    private Set<ProjectMemberDto> requests;
}

File: src/frontend/src/app/model/projectDto.ts
import {ProjectMemberDto} from './ProjectMemberDto';
import {UserProfileDto} from './UserProfileDto';

export class ProjectDto {
  id: number;
  name: string;
  projectKey: string;
  description: string;
  icon: any;
  owner: UserProfileDto;
  members: ProjectMemberDto[];
  requests: ProjectMemberDto[];
}

